api/v1beta1/azuremanagedmachinepool_types.go

```diff
@@ -520,6 +521,17 @@ type AzureManagedMachinePoolSpec struct {
 	// +optional
 	ScaleSetPriority *string `json:"scaleSetPriority,omitempty"`
 
+	// ScaleDownMode affects the cluster autoscaler behavior. Default to Delete. Possible values include: 'Deallocate', 'Delete'
+	// +kubebuilder:validation:Enum=Deallocate;Delete
+	// +kubebuilder:default=Delete
+	// +optional
+	ScaleDownMode *string `json:"scaleDownMode,omitempty"`
+
+	// SpotMaxPrice defines max price to pay for spot instance.
+	// Possible values are any decimal value greater than zero or -1 which indicates the willingness to pay any on-demand price.
+	// +optional
+	SpotMaxPrice *resource.Quantity `json:"spotMaxPrice,omitempty"`
```

Comment:
Can we get away with using `*float64` instead of using `*resource.Quantity` ?
Just a thought, not that I am against using `*resource.Quantity`. My question is rather a self clarifying thought to learn more about this decision.

Reply:
API conventions recommend against using floats due to risk during conversion https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types

> Avoid floating-point values as much as possible, and never use them in spec. Floating-point values cannot be reliably round-tripped (encoded and re-decoded) without changing, and have varying precision and representations across languages and architectures.

Reply:
Ah sweet, thanks for pinning it!